### ISSUE TITLE
[TA] report gnorm even when no clipping

### ISSUE THREAD

**Issue by stephenroller** [facebookresearch/ParlAI#1977]
Presently, gradient norms are only reported in TA when `--gradient-clip` is > 0.

https://github.com/facebookresearch/ParlAI/blob/c1e35bac34e7336b794f3ac48b100f9b17d14e86/parlai/core/torch_agent.py#L1800-L1807

https://github.com/facebookresearch/ParlAI/blob/c1e35bac34e7336b794f3ac48b100f9b17d14e86/parlai/core/torch_agent.py#L1005-L1006

However, this information is always useful. In the case of gradient clip being <= 0, the norm should be computed anyway. See the `torch.nn.utils.clip_grad_norm_` code to see how this is done.
